core/src/validator.rs

```diff
@@ -169,6 +171,40 @@ impl Validator {
             bank.slots_per_segment(),
         );
 
+        let working_bank = bank_forks.read().unwrap().working_bank();
```

Comment:
The overall logic here looks good but we'd need it to be turned inside out, in that the stake check and warning should all run in response to `set_handler()`.   The node may not have stake when it boots initially, but sometime later when Ctrl-C is pressed the node may have stake.